cmd/thv/app/client.go

```diff
@@ -0,0 +1,111 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stacklok/toolhive/cmd/thv/app/ui"
+	"github.com/stacklok/toolhive/pkg/client"
+	"github.com/stacklok/toolhive/pkg/config"
+)
+
+var clientCmd = &cobra.Command{
+	Use:   "client",
+	Short: "Manage MCP clients",
+	Long:  "The client command provides subcommands to manage MCP client integrations.",
+}
+
+var clientStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Show status of all supported MCP clients",
+	Long:  "Display the installation and registration status of all supported MCP clients in a table format.",
+	RunE:  clientStatusCmdFunc,
+}
+
+var clientSetupCmd = &cobra.Command{
+	Use:   "setup",
+	Short: "Interactively setup and register installed clients",
+	Long:  `Presents a list of installed but unregistered clients for interactive selection and registration.`,
+	RunE:  clientSetupCmdFunc,
+}
+
+func init() {
+	rootCmd.AddCommand(clientCmd)
+
+	clientCmd.AddCommand(clientStatusCmd)
+	clientCmd.AddCommand(clientSetupCmd)
+}
+
+func clientStatusCmdFunc(_ *cobra.Command, _ []string) error {
+	clientStatuses, err := client.GetClientStatus()
+	if err != nil {
+		return fmt.Errorf("failed to get client status: %w", err)
+	}
+	return ui.RenderClientStatusTable(clientStatuses)
+}
+
+func clientSetupCmdFunc(cmd *cobra.Command, _ []string) error {
+	clientStatuses, err := client.GetClientStatus()
+	if err != nil {
+		return fmt.Errorf("failed to get client status: %w", err)
+	}
+	availableClients := getAvailableClients(clientStatuses)
+	if len(availableClients) == 0 {
+		fmt.Println("All installed clients are already registered.")
```

Comment:
Nitpick: This message is misleading in the case where no clients are installed on the system. I would change it to something like `No new clients found` which seems applicable to the two scenarios.